src/blog/event-sourcing-vs-crud.md

```diff
@@ -0,0 +1,68 @@
+
+To be honest, event sourcing does generally require writing more code up front than CRUD. Lagom does not push event sourcing because it reduces the amount of code. Lagom pushes event sourcing because the problems that microservices introduce - such as ensuring consistency between services - are very easily solved when you are using event sourcing for persistence, and very hard to solve when you're using CRUD for persistence. A service using CRUD might have less code to start with, but as soon as it starts sharing data with other services, problems relating to handling failures and synchronisation of state become very difficult problems to solve, and will often require implementing complex solutions, including manual state and failure tracking. In fact, in my experience of using CRUD, dealing with these problems usually requires manually implementing event logs and asynchronous batch processes to handle them, so developers end up implementing both event sourcing and CRUD, which is far more code and far harder to maintain. In contrast, event sourcing inherently deals with these problems as a property of the architectural approach, with no special code implemented by developers.
+
+Put simply, using event sourcing over CRUD is an investment of more code and effort up front, but will yield a simpler, more resilient and easier to maintain system in the future. Lets take a look at how this looks in practice.
+
+## A blog system
+
+To demonstrate the difference between CRUD and event sourcing, we're going to use a blogging system as an example. A blogging system will have a service to manage blog posts, but then it will have other services, such as a search service for doing a natural text search on blog posts, a tag service for managing tags and returning blog posts by tag, and a recent post service for maintaining a list of recent blog posts to be rendered for example on the home page, or by an atom feed.
+
+A blog post starts in the unpublished state, may be edited a few times, and then at some point will be published. Once published, the search service, tag service and recent posts service need to notified about the blog post so that they can update their indexes.
+
+### CRUD code
+
+Here is what the CRUD example might look like if we were using JPA. Note we have left out getters/setters/equals/hashCode for brevity. First we have a JPA entity:
+
+@[entity](event-sourcing-vs-crud/jpa/Blog.java)
+
+Then we have some code that handles the creation of the blog post, editing the blog post, and publishing the blog post:
+
+@[operations](event-sourcing-vs-crud/jpa/BlogDao.java)
+
+### Event sourcing code
+
+Here is what the event sourcing example might look like using the Lagom persistence API.
```

Comment:
This writing tripped me. It looked like immediately below there should be a code snippet.